src/gun_http2.erl

```diff
@@ -934,6 +957,20 @@ keepalive(State=#http2_state{socket=Socket, transport=Transport, pings_unack=Pin
 			{Error, EvHandlerState}
 	end.
 
+ping(State=#http2_state{socket=Socket, transport=Transport, user_pings=UserPings}, StreamRef, ReplyTo) ->
+	%% Use non-zero 64-bit payload for user pings. 0 is reserved for keepalive.
+	Payload = case erlang:monotonic_time(microsecond) band 16#ffffffffffffffff of
+		0 -> 1;
+		Payload0 -> Payload0
+	end,
```

Comment:
Just have a counter that starts at 1 (or 1001, or other...) and increment every time, it'll save us trouble.

Reply:
I guess I wanted avoid wasting memory to store a counter for a feature that almost nobody will use.

What trouble do you anticipate?

Reply:
Future changes may use more than 0 for keepalive and we don't want conflicts to happen. The counter can be stored in `user_pings` when user pings are used, and `user_pings` can be set to `false` when they are not used.

Reply:
`erlang:unique_integer([monotonic, positive])` is probably good enough.

Reply:
Sure, that's fine too. Which do you prefer: unique integer or counter?

Reply:
The unique integer function is a kind of counter. It will work better than keeping a counter ourselves.

``` erlang
1> erlang:unique_integer([monotonic, positive]).
1
2> erlang:unique_integer([monotonic, positive]).
2
3> erlang:unique_integer([monotonic, positive]).
3
4> erlang:unique_integer([monotonic, positive]).
4
```

Reply:
Sure. Unique integer does some inter-process sync so it may be slower but that's not a concern here. This is not a hot code patch.

Reply:
Same as monotonic_time in that regard, but better control over the values it returns.

Make it send payloads starting at 10000 (`10000 + erlang:unique_integer([monotonic, positive])`) so there's plenty of space left if we need other values in the future. Not sure we need to `band 16#ffffffffffffffff`, I don't know how long that would take for unique_integer to reach that value, but if we do the wrapped value should be > 10000.